src/main/java/com/jvm_bloggers/core/github/GithubAuthenticationFilter.java

```diff
@@ -0,0 +1,29 @@
+package com.jvm_bloggers.core.github;
+
+import org.apache.wicket.util.string.Strings;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.core.HttpHeaders;
+
+@Component
+public class GithubAuthenticationFilter implements ClientRequestFilter {
+
+    private final GithubProperties githubProperties;
+
+    public GithubAuthenticationFilter(GithubProperties githubProperties) {
```

Comment:
This handcrafted constructor can be replaced with [`@RequiredArgsConstructor`](https://projectlombok.org/api/lombok/RequiredArgsConstructor.html) annotation.